EndlessClient/Dialogs/BarberDialog.cs

```diff
@@ -0,0 +1,209 @@
+                            IEIFFileProvider eifFileProvider,
+                            ISfxPlayer sfxPlayer)
+            : base(nativeGraphicsManager, isInGame: true)
+        {
+            _characterRepository = characterLvRepository;
+            _dialogIconService = dialogIconService;
+            _localizedStringFinder = localizedStringFinder;
+            _barberActions = barberActions;
+            _characterInventoryProvider = characterInventoryProvider;
+            _messageBoxFactory = messageBoxFactory;
+            _eifFileProvider = eifFileProvider;
+            _sfxPlayer = sfxPlayer;
+
+            BackgroundTexture = GraphicsManager.TextureFromResource(GFXTypes.PostLoginUI, 56);
+
+            var mainCharacterRenderProperties = _characterRepository.MainCharacter.RenderProperties;
+            _characterControl = new CreateCharacterControl(mainCharacterRenderProperties, rendererFactory)
+            {
+                DrawPosition = new Vector2(210, 19),
+            };
+
+            InitializeCharacterControl();
+            InitializeDialogItems(dialogButtonService);
+            CenterInGameView();
+        }
+
+        private void InitializeCharacterControl()
+        {
+            _characterControl.SetParentControl(this);
+        }
+
+        private void InitializeDialogItems(IEODialogButtonService dialogButtonService)
+        {
+            var cancel = CreateButton(dialogButtonService, new Vector2(215, 151), SmallButton.Cancel);
+            cancel.OnClick += (_, _) => Close(XNADialogResult.Cancel);
+
+            CreateListItems();
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _characterControl.Initialize();
+            UpdateListItems(_characterRepository.MainCharacter.RenderProperties);
+        }
+
+        private void CreateListItems()
+        {
+            _changeHairItem = CreateListDialogItem(DialogIcon.BarberHairModel, 25, ChangeHairStyle_Click);
+            _changeHairColor = CreateListDialogItem(DialogIcon.BarberChangeHairColor, 60, ChangeHairColor_Click);
+            _changeBuyHairStyleOrColor = CreateListDialogItem(DialogIcon.BarberOk, 95, BuyHair_Click);
+
+            ApplyWidthOverride(_changeHairItem);
+            ApplyWidthOverride(_changeHairColor);
+            ApplyWidthOverride(_changeBuyHairStyleOrColor);
+
+            ApplyHighlightOffsetXOverride(_changeHairItem);
+            ApplyHighlightOffsetXOverride(_changeHairColor);
+            ApplyHighlightOffsetXOverride(_changeBuyHairStyleOrColor);
+        }
+
+        private void ApplyWidthOverride(ListDialogItem item)
+        {
+            item.WidthOverride = AdjustedWidth;
+            item.SetItemSize(item.WidthOverride.GetValueOrDefault(232), item.Style == ListDialogItem.ListItemStyle.Large ? 36 : 13); 
+        }
+
+        private void ApplyHighlightOffsetXOverride(ListDialogItem item)
+        {
+            item.OffsetX = AdjustedHighlightXOffset;
```

Comment:
I appreciate that you're trying to make the code more readable by splitting out a bunch of helper methods. I don't know if these are really adding a lot of clarity though, especially for these one-line things. IMO it's probably better to just initialize the list items in-line so readers (me) don't have to decode the call stack and trace through what's actually happening between all these calls.

I'll leave it up to your discretion on this one (comment here with your decision so I know which direction you think is better), but I think just constructing the list items in one spot with a property initializer list would make this a lot easier to parse through.

Reply:
"every abstraction has its cost (readability in this case)". I'll go inline.